contracts/ligo/src/english_auction/english_auction_fa2.mligo

```diff
@@ -0,0 +1,231 @@
+  {
+    fa2_address : address;
+    fa2_batch : (fa2_tokens list);
+  }
+
+type bid_currency = 
+  [@layout:comb]
+  {
+      fa2_address : address;
+      token_id : token_id;
+  }
+
+type auction =
+  [@layout:comb]
+  {
+    seller : address;
+    current_bid : nat;
+    start_time : timestamp;
+    last_bid_time : timestamp;
+    round_time : int;
+    extend_time : int;
+    asset : (tokens list);
+    min_raise_percent : nat;
+    min_raise : nat;
+    end_time : timestamp;
+    highest_bidder : address;
+  }
+
+type configure_param =
+  [@layout:comb]
+  {
+    opening_price : nat;
+    min_raise_percent : nat;
+    min_raise : nat;
+    round_time : nat;
+    extend_time : nat;
+    asset : (tokens list);
+    start_time : timestamp;
+    end_time : timestamp;
+  }
+
+type bid_param = 
+  [@layout:comb]
+  {
+    asset_id : nat;
+    bid_amount : nat;
+  }
+
+type auction_entrypoints =
+  | Configure of configure_param
+  | Bid of bid_param
+  | Cancel of nat
+  | Resolve of nat
+  | Admin of pauseable_admin
+
+type storage =
+  [@layout:comb]
+  {
+    pauseable_admin : pauseable_admin_storage;
+    current_id : nat;
+    max_auction_time : nat;
+    max_config_to_start_time : nat;
+    bid_currency : bid_currency;
+    auctions : (nat, auction) big_map
+  }
+
+type return = operation list * storage
+
+let assert_msg (condition, msg : bool * string ) : unit =
+  if (not condition) then failwith(msg) else unit
+
+let address_to_contract_transfer_entrypoint(add : address) : ((transfer list) contract) =
+  let c : (transfer list) contract option = Tezos.get_entrypoint_opt "%transfer" add in
+  match c with
+    None -> (failwith "Invalid FA2 Address" : (transfer list) contract)
+  | Some c ->  c
+
+let fa2_batch_to_transfer_param(fa2_batch, from_, to_ : fa2_tokens list * address * address) : (transfer list) =
+  let to_tx (fa2_tokens : fa2_tokens) : transfer_destination = {
+      to_ = to_;
+      token_id = fa2_tokens.token_id;
+      amount = fa2_tokens.amount;
+  } in
+  let txs = List.map to_tx fa2_batch in
+  let transfer_param = {from_ = from_; txs = txs} in
+  [transfer_param]
+
+(*Transfers tokens in batch from a single FA2 contract*)
+let tokens_to_operation(from_ : address) (to_ : address) (tokens : tokens): operation =
+  let param = fa2_batch_to_transfer_param(tokens.fa2_batch, from_, to_) in
+  let c = address_to_contract_transfer_entrypoint(tokens.fa2_address) in
+  (Tezos.transaction param 0mutez c)
+
+let fa2_fee_transfer (from_, to_, qty, bid_currency : address * address * nat * bid_currency) : operation = 
+  let tx_param : tokens = 
+    ({
+        fa2_address = bid_currency.fa2_address;
+        fa2_batch = [({token_id = bid_currency.token_id; amount = qty} : fa2_tokens)]
+    }) in
+  let op = tokens_to_operation from_ to_ tx_param in
+  op
+
+(*Handles transfers of tokens across FA2 Contracts*)
+let tokens_to_operation_list((tokens_list, from_, to_) : tokens list * address * address) : (operation list) =
```

Comment:
1. Consider a better name, something like `transfer_tokens`
2. Consider replacing `tokens_to_operation_list` and `fa2_fee_transfer` into a single `transfer_tokens` function